### PR TITLE
Deprecate X509_certificate_type

### LINKED ISSUE #13997: Deprecate X509_certificate_type as it is not extensible and provider aware
The function returns EVP_PK* flags not used anywhere else. Due to this API contract it is not extensible in regards to providers and possibly unknown algorithms.

It should be deprecated for 3.0. No replacement is needed as this is just a convenience function and the information it provides can be obtained via X509_get0_pubkey and X509_get0_signature.

### PR DESCRIPTION
**Pull request by richsalz**
This was easy; let's see if the CI's make it. :)

Fixes #13997
ping @t8m

### CODE REVIEW

File: crypto/x509/build.info
Comment (t8m) [inline crypto/x509/build.info:19]: `!$disabled...`

Comment (richsalz) [reply crypto/x509/build.info]: :)

File: CHANGES.md
Comment (t8m) [inline CHANGES.md:26]: Typo: undocumended

**Review by t8m — COMMENTED** [review]
Just one typo, otherwise LGTM.

File: CHANGES.md
Comment (richsalz) [reply CHANGES.md]: udpated commit pushed thanks.

**Comment by richsalz** [comment]
Build failure is unrelated:
```
   # CMD: openssl cms -provider-path ../../providers -config ../../test/fips-and-base.cnf -provider fips -encrypt -in ../../test/smcont.txt -stream -out cms2cms-mod-13.cms -recip ../../test/smime-certs/smdh.pem -aes128
Using configuration from ../../test/fips-and-base.cnf
../../util/wrap.pl ../../apps/openssl cms -provider-path ../../providers -config ../../test/fips-and-base.cnf -provider fips -encrypt -in ../../test/smcont.txt -stream -out cms2cms-mod-13.cms -recip ../../test/smime-certs/smdh.pem -aes128 => 0
    # CMD: openssl cms -provider-path ../../providers -config ../../test/fips-and-base.cnf -provider fips -decrypt -recip ../../test/smime-certs/smdh.pem -in cms2cms-mod-13.cms -out cms2cms-mod-13.txt
Using configuration from ../../test/fips-and-base.cnf
Error decrypting CMS using private key
806BA35D0F7F0000:error:1700006C:CMS routines:cms_get0_signed:content type not signed data:crypto/cms/cms_sd.c:30:
806BA35D0F7F0000:error:02800066:Diffie-Hellman routines:dh_buf2key:invalid public key:crypto/dh/dh_key.c:392:
806BA35D0F7F0000:error:170000BC:CMS routines:dh_cms_decrypt:peer key error:crypto/cms/cms_dh.c:173:
../../util/wrap.pl ../../apps/openssl cms -provider-path ../../providers -config ../../test/fips-and-base.cnf -provider fips -decrypt -recip ../../test/smime-certs/smdh.pem -in cms2cms-mod-13.cms -out cms2cms-mod-13.txt => 4
    not ok 13 - enveloped content test streaming S/MIME format, X9.42 DH
# ------------------------------------------------------------------------------
    # Looks like you failed 1 test of 14.
not ok 5 - CMS <=> CMS consistency tests, modified key parameters
```

**Comment by slontis** [comment]
Wonder if this is an intermittent problem?.. Not seeing it on later builds.

**Comment by t8m** [comment]
The intermittent CI failure is fixed by #13958

**Review by t8m — APPROVED** [review]
LGTM

**Comment by t8m** [comment]
Ping @openssl/committers for the second review

**Review by DDvO — APPROVED** [review]
LGTM

**Comment by t8m** [comment]
Merged to master. Thank you for the contribution!

**Pull request closed without merging**
